camera_info_manager/src/camera_info_manager.cpp

```diff
@@ -281,9 +281,10 @@ bool CameraInfoManager::loadCalibrationFile(const std::string &filename,
         }
       success = true;
       {
-        // lock only while updating cam_info_
+        // lock only while updating cam_info_ and url_
         boost::mutex::scoped_lock lock(mutex_);
         cam_info_ = cam_info;
+        url_ = filename;
```

Comment:
Note that the actual URL may have been a ``package:`` reference. If so, this would convert it to a ``file:`` reference to that package's location in the current workspace. That may not always be what you wanted.